version: '3.3'

services:
  namenode:
    image: hadoop-namenode
    networks:
      - hbase
    volumes:
      - namenode:/hadoop/dfs/name
    environment:
      - CLUSTER_NAME=test
    env_file:
      - ./hadoop.env
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        condition: on-failure

  datanode:
    image: hadoop-datanode
    networks:
      - hbase
    volumes:
      - datanode:/hadoop/dfs/data
    environment:
      SERVICE_PRECONDITION: "namenode:50070"
    deploy:
      mode: global
      restart_policy:
        condition: on-failure

  resourcemanager:
    image: hadoop-resourcemanager
    networks:
      - hbase
    environment:
      SERVICE_PRECONDITION: "namenode:50070 datanode:50075"
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        condition: on-failure
    healthcheck:
      disable: true

  nodemanager:
    image: bde2020/hadoop-nodemanager:2.0.0-hadoop3.1.1-java8
    networks:
      - hbase
    environment:
      SERVICE_PRECONDITION: "namenode:50070 datanode:50075 resourcemanager:8088"
    deploy:
      mode: global
      restart_policy:
        condition: on-failure

  historyserver:
    image: hadoop-historyserver
    networks:
      - hbase
    volumes:
      - hadoop_historyserver:/hadoop/yarn/timeline
    environment:
      SERVICE_PRECONDITION: "namenode:50070 datanode:50075 resourcemanager:8088"
    deploy:
      mode: replicated
      replicas: 1

volumes:
  datanode:
  namenode:
  hadoop_historyserver:

networks:
  hbase:
    external:
      name: hbase
